File: scratchpad/14. Longest Common Prefix.cpp
#include <bits/stdc++.h>

class Solution {
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        // Bruteforce approach
        int min_len = INT_MAX;

        std::string res = "";

        for (const auto& str : strs) {
            min_len = std::min(min_len, (int)str.length());
        }    
        
        std::string arb_str = strs[0];
        for (int i = 0; i < min_len; i++) {
            for (const auto& str : strs) {
                if (str[i] != arb_str[i])
                    return res;
            }
            res += arb_str[i];
        }
        return res;
    }
};

int main() {
    std::vector<std::string> strs {"flower","flow","flight"};
    std::cout << (new Solution())->longestCommonPrefix(strs) << "\n";
}

File: strings/14. Longest Common Prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Bruteforce approach
        res = "" 

        min_len = float('inf')
        for str in strs:
            min_len = min(min_len, len(str))

        arb_str = strs[0]
        for i in range(min_len):
            for str in strs:
                if str[i] != arb_str[i]:
                    return res 
            res += str[i]
        
        return res


if __name__ == "__main__":
    print(Solution().longestCommonPrefix( ["flower","flow","flight"]))
